backend/src/apiserver/storage/job_store.go

```diff
@@ -490,10 +490,12 @@ func (s *JobStore) UpdateJob(swf *util.ScheduledWorkflow) error {
 }
 
 // factory function for job store.
-func NewJobStore(db *DB, time util.TimeInterface) *JobStore {
+func NewJobStore(db *DB, time util.TimeInterface, pipelineStore PipelineStoreInterface) *JobStore {
+	// If pipelineStore is specified and it not-nil, it will be used instead of db.
+	// This will make pipelines and pipeline versions to get stored in K8s instead of Database.
```

Comment:
```suggestion
// If pipelineStore is provided, it will be used instead of direct database queries for getting pipelines
// and pipeline versions.
func NewJobStore(db *DB, time util.TimeInterface, pipelineStore PipelineStoreInterface) *JobStore {
```